staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -550,7 +550,7 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(
 				n.Labels = map[string]string{}
 			}
 
-			k8sNamespaceRegex := regexp.MustCompile("(kubernetes|k8s).io/")
+			k8sNamespaceRegex := regexp.MustCompile(`(kubernetes|k8s)\.io/`)
```

Comment:
if we want to be fully correct, we'll also make sure the domain portion is exactly kubernetes.io or k8s.io, or has a leading dot making it a subdomain

```suggestion
			k8sNamespaceRegex := regexp.MustCompile(`(^|\.)(kubernetes|k8s)\.io/`)
```

Reply:
+1, this seems prudent to me.